chain/network/src/peer_manager/peer_manager_actor.rs

```diff
@@ -423,32 +396,24 @@ impl PeerManagerActor {
     }
 
     fn broadcast_accounts(&mut self, mut accounts: Vec<AnnounceAccount>) {
-        // Filter the accounts again, so that we're sending only the ones that were not added.
-        // without it - if we have multiple 'broadcast_accounts' calls queued up, we'll end up sending a lot of repeated messages.
-        accounts.retain(|announce_account| {
-            match self.routing_table_view.get_announce(&announce_account.account_id) {
-                Some(current_announce_account)
-                    if announce_account.epoch_id == current_announce_account.epoch_id =>
-                {
-                    false
-                }
-                _ => true,
-            }
-        });
-
+        // Filter the accounts, so that we're sending only the ones that were not added.
+        accounts.retain(|a| !self.routing_table_view.contains_account(&a));
         if accounts.is_empty() {
             return;
         }
         debug!(target: "network", account_id = ?self.config.validator.as_ref().map(|v|v.account_id()), ?accounts, "Received new accounts");
-        for account in accounts.iter() {
+        for account in &accounts {
             self.routing_table_view.add_account(account.clone());
         }
 
         Self::broadcast_message(
             self.network_metrics.clone(),
             &self.connected_peers,
             SendMessage {
-                message: PeerMessage::SyncRoutingTable(RoutingTableUpdate::from_accounts(accounts)),
+                message: PeerMessage::SyncRoutingTable(RoutingTableUpdate {
+                    accounts,
+                    ..Default::default()
+                }),
```

Comment:
very weak opinion, but I kinda dislike the `..Default::default()` pattern, seeing it makes me feel like something's missing. Here, perhaps we want a ctor fn? `RountingTableUpdate::new_accounts(acconts)? `